clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -3640,6 +3655,24 @@ void SemaHLSL::ActOnVariableDeclarator(VarDecl *VD) {
 
     // process explicit bindings
     processExplicitBindingsOnDecl(VD);
+
+    if (VD->getType()->isHLSLResourceRecordArray()) {
+      // If the resource array does not have an explicit binding attribute,
+      // create an implicit one. It will be used to transfer implicit binding
+      // order_ID to codegen.
+      if (!VD->hasAttr<HLSLVkBindingAttr>()) {
```

Comment:
Shouldn't this check if it's missing HLSLResourceBindingAttr? Or is this saying that HLSLVkBindingAttr is only added when a binding attribute is explicitly spelled out?